backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -902,7 +879,19 @@ public Project setProjectShortName(String projectId, String projectShortName, Us
   private List<ProjectEntity> getProjectEntityByBatchIds(Session session, List<String> projectIds) {
     Query query = session.createQuery(GET_PROJECT_BY_IDS_HQL);
     query.setParameterList("ids", projectIds);
-    return query.list();
+    final List<ProjectEntity> projectEntities = query.list();
+    return projectEntities.stream()
+        .map(
+            projectEntity -> {
+              ResourceVisibility projectVisibility =
+                  roleService.getProjectVisibility(
+                      projectEntity.getId(),
+                      projectEntity.getWorkspace(),
+                      projectEntity.getWorkspace_type());
+              projectEntity.setProjectVisibility(projectVisibility);
```

Comment:
We need to set workspace and owner too, right?

Reply:
Yes, I will need to change this to fix that.  I am currently relying upon the DB values in MDB, but those aren't the correct source of truth now.